yt_dlp/extractor/weibo.py

```diff
@@ -1,134 +1,246 @@
+    _VALID_URL = r'https://weibo.com/tv/show/(?P<id>\d+:\d+)'
+    _TESTS = [{
+        'url': 'https://weibo.com/tv/show/1034:4797699866951785?from=old_pc_videoshow',
+        'info_dict': {
+            'id': '4797700463137878',
+            'ext': 'mp4',
+            'display_id': 'LEZDodaiW',
+            'title': '呃，稍微了解了一下靡烟miya，感觉这东西也太二了',
+            'description': '呃，稍微了解了一下靡烟miya，感觉这东西也太二了 http://t.cn/A6aerGsM ​​​',
+            'duration': 76,
+            'timestamp': 1659344278,
+            'upload_date': '20220801',
+            'thumbnail': r're:https://.*\.jpg',
+            'uploader': '君子爱财陈平安',
+            'uploader_id': '3905382233',
+            'uploader_url': 'https://weibo.com/u/3905382233',
+            'view_count': int,
+            'like_count': int,
+            'repost_count': int,
         }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        post_data = f'data={{"Component_Play_Playinfo":{{"oid":"{video_id}"}}}}'.encode()
+        video_info = self._weibo_download_json(
+            f'https://weibo.com/tv/api/component?page=%2Ftv%2Fshow%2F{video_id.replace(":", "%3A")}',
+            video_id, headers={'Referer': url}, data=post_data)['data']['Component_Play_Playinfo']
+        return self.url_result(f'https://weibo.com/0/{video_info["mid"]}', WeiboIE)
+
+
+class WeiboUserIE(WeiboBaseIE):
+    _VALID_URL = r'https://weibo.com/u/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://weibo.com/u/2066652961?tabtype=video',
+        'info_dict': {
+            'id': '2066652961',
+            'title': '萧影殿下的视频',
+            'description': '萧影殿下的全部视频',
+            'uploader': '萧影殿下',
+        },
+        'playlist_mincount': 195,
+    }]
+
+    def _entries(self, uid):
+        query = {'uid': uid, 'cursor': 0}
+        for page in itertools.count(1):
+            response = self._weibo_download_json(
+                'https://weibo.com/ajax/profile/getWaterFallContent',
+                uid, query=query, note=f'Downloading videos page {page}')['data']
+            for video_info in response.get('list', []):
+                yield self._parse_video_info(video_info)
+            if (int_or_none(response.get('next_cursor')) or -1) > 0:
+                query['cursor'] = response['next_cursor']
+            else:
+                break
+
+    def _first_and_entries(self, uid):
+        entries = self._entries(uid)
+        video = next(entries)
+
+        def _entries():
+            yield video
+            yield from entries
+        return video, _entries()
+
+    def _real_extract(self, url):
+        uid = self._match_id(url)
+        video, entries = self._first_and_entries(uid)
+        return self.playlist_result(entries, uid, **traverse_obj(video, {
+            'title': ('uploader', {lambda i: f'{i}的视频' if i else None}),
+            'description': ('uploader', {lambda i: f'{i}的全部视频' if i else None}),
+            'uploader': 'uploader',
+        }))
```

Comment:
The multiple `entries`/`_entries` var/functions make this hard to read for me.

It could be simplified somewhat and we could eliminate a duplicated first page request like this:
```suggestion
    def _fetch_page(self, uid, cursor=0, page=1):
        return self._weibo_download_json(
            'https://weibo.com/ajax/profile/getWaterFallContent',
            uid, note=f'Downloading videos page {page}',
            query={'uid': uid, 'cursor': cursor})['data']

    def _entries(self, uid, first_page):
        cursor = 0 
        for page in itertools.count(1):
            response = first_page if page == 1 else self._fetch_page(uid, cursor, page)
            for video_info in traverse_obj(response, ('list', ..., {dict})):
                yield self._parse_video_info(video_info)
            cursor = response.get('next_cursor')
            if (int_or_none(cursor) or -1) < 0:
                break

    def _real_extract(self, url):
        uid = self._match_id(url)
        first_page = self._fetch_page(uid)

        return self.playlist_result(
            self._entries(uid, first_page), uid, **traverse_obj(first_page, {
                'title': ('uploader', {lambda i: f'{i}的视频' if i else None}),
                'description': ('uploader', {lambda i: f'{i}的全部视频' if i else None}),
                'uploader': 'uploader',
            }))
```
What do you think?

Reply:
I agree it is more readable this way.